docker-compose.yml

```diff
@@ -1,4 +1,17 @@
-app:
-  build: .
-  volumes:
-  - .:/go/src/github.com/bitrise-io/bitrise
+services:
+  main-container:
+    build:
+      context: .
+    platform: linux/amd64
+    environment:
+      BITRISE_DOCKER_MOUNT_OVERRIDES: /Users/[yourusername]/.ssh:/root/.ssh,./_containers/.bitrise:/root/.bitrise,./:/usr/src/bitrise,./_local:/usr/local/bundle,./_tmp:/usr/src/bitrise/_tmp,/tmp:/tmp
+    volumes:
+    - /Users/[yourusername]/.ssh:/root/.ssh
+    - /tmp:/tmp
+    - ./_containers/.bitrise:/root/.bitrise
+    - ./:/usr/src/bitrise
+    - ./_tmp:/usr/src/bitrise/_tmp
+    - /var/run/docker.sock:/var/run/docker.sock
+    - ./_local:/usr/local/bundle
```

Comment:
cc: @bvatai-br This compose file will probably confuse future readers, especially the bundle folder I believe we only needed for the ruby (subscription) POC. We should make it so it contains the bare minimum mounts and config for a simple script step IMO and maybe comments about these gotchas such as the ssh key (only needed when you want to clone stuff).